[NatLeong] iP

NatLeong opened the pull request:
### Dav TaskBot 

> “For every minute spent organizing, an hour is earned." - [source](https://www.sortly.com/blog/6-inspiring-quotes-for-getting-organized-manage-and-maintain-business-better/)

Dav is here to make your life easier! Dav is
- _EASY_ to use 
- ~~Fast~~  Efficient
- Text-based 

**Types of Tasks supported** :spiral_notepad:
1. To do 
2. Event 
3. Deadline

**All** type of tasks for **all** needs...

**Features:**
- [ ] Store Tasks
- [ ] Keep track of Task status 
- [ ] Search for Tasks
- [ ] Tasks date and time information 
***
This is how Dav runs. Here's the `main` method 
```java
public static void main(String[] args) {
    new Dav().run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Add Level 0

Commit:
Add Level 1

Commit:
Add Level 2

Commit:
Add Level 3

Commit:
Add Level 4

Commit:
Add A-TextUiTesting

Commit:
Add Level-5

Commit:
Add Level 6

Commit:
Add revised A-TextUiTesting

Commit:
Add Level 8

Commit:
Merge branch 'branch-Level-8'

Commit:
Add A-MoreOOP

Commit:
Merge branch 'branch_A-MoreOOP'

Commit:
Add A-Packages

Commit:
Merge branch 'branch_A-Packages'

Commit:
Changes for packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add A-Gradle

Commit:
Merge branch 'branch_A-Gradle'

Commit:
Add A-JUnit

Commit:
Merge branch 'branch_A-JUnit'

dhlee03 commented on src/main/java/dav/Parser.java:
```diff
@@ -0,0 +1,39 @@
+    public static void parseUserInput(String input, TaskList tasks, Ui ui, Storage storage) {
+        System.out.println("____________________________________________________________");
+
+        try {
```
As there are many commands, using switch might look better.

dhlee03 commented on src/main/java/dav/TaskList.java:
```diff
@@ -0,0 +1,175 @@
+
+    public void addTodoTask(String taskDescription) {
+        if (taskDescription.isEmpty()) {
+            System.out.println("Do nothing?");
```
This can be placed in UI

dhlee03 commented on src/main/java/dav/TaskList.java:
```diff
@@ -0,0 +1,175 @@
+                System.out.println("Invalid event task format.");
+            }
+        } else {
+            System.out.println("Invalid event task format.");
```
Nice error handling

dhlee03 reviewed (commented):
Overall the code looks good to me👍. Few minor changes needed.

taniatsq commented on src/main/java/dav/DeadlineTask.java:
```diff
@@ -0,0 +1,42 @@
+package dav;
+import java.time.LocalDateTime;
```
Can be more consistent with structure of import statements and package statements. Leave spacing between them and between the imports and public class

taniatsq commented on src/main/java/dav/Task.java:
```diff
@@ -0,0 +1,46 @@
+        return "";
+    }
+
+    public static Task parseTask(String data) {
```
nice use of subclass overridden methods here

taniatsq reviewed (commented):
Overall, quite easy to understand the code but adding javadocs would definitely be helpful.

Commit:
Add JavaDoc comments

Commit:
To Coding Standards

Commit:
Level-9 increment

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Address Deeply nested methods

Commit:
Level 10 increment

Commit:
Merge branch 'branch-Level-10'

Commit:
The existing code for parsing user input in the task management system lacks robust validation for various commands, potentially leading to unexpected behavior or errors.

To enhance the reliability and user experience, it is crucial to strengthen the input validation mechanism. This will prevent incorrect commands and ensure that the system behaves as expected.

Introduced more comprehensive input validation checks for 'todo,' 'deadline,' 'event,' 'check,' 'delete,' 'find,' 'mark,' and 'unmark' commands. Assertions have been added to verify the length of input for each command, helping catch format-related issues during development and testing.

Assertions are used to enforce specific conditions during development, acting as a safety net to catch potential issues early in the development process. These checks aim to ensure that the input adheres to the expected format for each command, promoting a more robust and error-resistant codebase.

Commit:
The existing codebase has areas where code quality can be improved, including readability, maintainability, and adherence to best practices.

Improving code quality is essential for long-term maintainability, collaboration, and reducing the likelihood of introducing bugs. It enhances the overall development experience and facilitates future enhancements.

Performed a comprehensive code review and refactored sections of the code to adhere to coding standards, enhance readability, and promote consistency. Addressed code smells, improved Class names, and introduced comments where necessary. Ensured that the code follows established best practices for Java development.

The changes are made in a systematic and focused manner, addressing specific aspects of code quality identified during the code review. The goal is to make the code more understandable, maintainable, and aligned with established coding conventions.

NatLeong committed:
Merge pull request #2 from NatLeong/branch-A-Assertions

A-Assertions

NatLeong committed:
Merge pull request #3 from NatLeong/branch-A-CodeQuality

A-CodeQuality

Commit:
No fetaure to tag tasks.

Tagging of tasks increase functionality of task bot

Refactored the Task class to include methods for adding and removing tags.

This improves the user experience by conveying relevant information about tag handling.

Commit:
Merge branch 'branch-BCD-Extension'

Commit:
Add more JUnit Tests and updated Junit tests to test for tags

Commit:
Improve GUI for better aesthetics and usability

Commit:
Merge branch 'branch-A-BetterGUI'

Commit:
Add more personality to Dav

Changed some responses to be more enthusiastic

Commit:
Merge branch 'branch-A-Personality'

Commit:
Added User guide and Ui screenshot

Commit:
Merge branch 'branch-A-UserGuide'

Commit:
Added creation of new file if file does not exist.

Commit:
Improved command format error messages.

Commit:
Improved code quality

Commit:
Improved ReadMe and Ui screenshot.

Commit:
Decreased Ui size

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md
